strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-yum-layout-provider/src/test/java/org/carlspring/strongbox/artifact/coordinates/RpmArtifactCoordinatesTest.java

```diff
@@ -0,0 +1,102 @@
+package org.carlspring.strongbox.artifact.coordinates;
+
+import org.carlspring.strongbox.domain.RpmPackageType;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.RepetitionInfo;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+/**
+ * Test class for testing functionality of parsing, creating and validating Rpm artifact coordinates
+ * @author Ilya Shatalov <ilya@alov.me>
+ */
+class RpmArtifactCoordinatesTest {
+
+    private static ArrayList<ArrayList<String>> parsedPackages = new ArrayList<>();
+
+    private static ArrayList<String> packageExamples = new ArrayList<>();
+
+
+    @BeforeAll
+    public static void populatePackages()
+    {
+        packageExamples.add("libqxt-qt5-0.7.0-0.15.20130718giteaf6872f6ad4.fc30.x86_64.rpm");
+        packageExamples.add("atinject-1-28.20100611svn86.module_f28+3939+dc18cd75.noarch.rpm");
+        packageExamples.add("apache-commons-lang3-3.7-3.module_f28+3939+dc18cd75.noarch.rpm");
+        packageExamples.add("wingpanel-indicator-sound-2.1.3-1.fc30.x86_64.rpm");
+        packageExamples.add("libglvnd-1.1.0-4.gitf92208b.fc30.i686.rpm");
+        packageExamples.add("apache-commons-lang3-3.7-3.module_f28+3939+dc18cd75.noarch.rpm");
+        packageExamples.add("libglvnd-1.1.0-4.gitf92208b.fc30.i686.rpm");
+        packageExamples.add("totem-3.32.1-1.fc30.x86_64.rpm");
+        packageExamples.add("libsigc++20-2.10.2-1.fc30.x86_64.rpm");
+        packageExamples.add("egl-wayland-1.1.3-1.fc30.x86_64.rpm");
+        packageExamples.add("urw-base35-d050000l-fonts-20170801-12.fc30.noarch.rpm");
+        packageExamples.add("libbabeltrace-1.5.6-2.fc30.x86_64.rpm");
+        packageExamples.add("redhat-lsb-submod-security-4.1-47.fc30.x86_64.rpm");
+        packageExamples.add("cdi-api-1.2-8.module_f28+3939+dc18cd75.noarch.rpm");
+        packageExamples.add("libstdc++-9.2.1-1.fc30.src.rpm");
+        packageExamples.add("gmp-c++-6.1.2-10.fc30.src.rpm");
```

Comment:
This looks like a very good list of artifact coordinates to test the parsing against! Thanks! :)